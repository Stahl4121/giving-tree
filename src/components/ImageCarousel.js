import React from 'react';

import { makeStyles } from '@material-ui/core/styles';
import MobileStepper from '@material-ui/core/MobileStepper';
import Button from '@material-ui/core/Button';
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import CardMedia from '@material-ui/core/CardMedia';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles({
  root: {
    maxWidth: 500,
    flexGrow: 1,
    height: 35,
  },
  narrow: {
    maxWidth: 500,
  },
  cardMedia: {
    height: 250,
  },
});

export default function ImageCarousel(props) {
  const classes = useStyles();
  const [activeStep, setActiveStep] = React.useState(0);

  const handleNext = () => {
    setActiveStep(prevActiveStep => prevActiveStep + 1);
  };

  const handleBack = () => {
    setActiveStep(prevActiveStep => prevActiveStep - 1);
  };

  return (
    <div className={classes.narrow}>
      <CardMedia
        className={classes.cardMedia}
        image={props.img[activeStep]}
        title="Grant Image"
      />
      <div>
        <Grid
          container
          direction='row'
          justify='space-between'
          alignItems='center'
        >
          {props.img.length > 1 &&
            <MobileStepper
              variant='dots'
              steps={props.img.length}
              position='static'
              activeStep={activeStep}
              className={classes.root}
              nextButton={
                <Button size='small' onClick={handleNext} disabled={activeStep === props.img.length - 1}>
                  Next
                <KeyboardArrowRight />
                </Button>
              }
              backButton={
                <Button size='small' onClick={handleBack} disabled={activeStep === 0}>
                  <KeyboardArrowLeft />
                  Back
              </Button>
              }
            />
          }
        </Grid>
      </div>
    </div>
  );
}