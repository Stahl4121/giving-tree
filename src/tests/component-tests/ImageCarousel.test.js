
import React from "react";
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import ImageCarousel from "../../components/ImageCarousel.js";

describe('Component: ImageCarousel', () => {
  var enzyme = require('enzyme');
  enzyme.configure({ adapter: new Adapter() });

  it('Renders correctly', () => {
    const mockImg = '../../iconSmall.png';
    
    const wrap = enzyme.mount(<ImageCarousel img={mockImg}/>)
    expect(wrap).toMatchSnapshot();
  });

})
