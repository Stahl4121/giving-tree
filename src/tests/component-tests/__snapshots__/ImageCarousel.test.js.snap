// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: ImageCarousel Renders correctly 1`] = `ReactWrapper {}`;

exports[`Image Carousel basics Renders correctly 1`] = `ShallowWrapper {}`;
